# Data Expo 2009 - Airline on-time performance Analysis
By Abdullah Almuzaini

## Table of Contents
- [Introduction](#intro)
- [Part I - Data Wrngling](#wrnagling)
- [Part II - Univariate Data Exploration](#unidata)
- [Part III - Bivariate Data Exploration](#bivData)
- [Part IV - Multivariate Data Exploration](#multivData)


 <a id='intro'>   </a>
## Introduction

<br><br>
<a id='wrnagling'>   </a>
## Part I - Data Wrngling


In [1]:
# The first step is to import all needed libraryies
# This cell is devote specificlly for importing libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline



<p style="font-size:18px"> The original dataset folder contains a dataset about flights for each year since 1987 to 2008. However, for this project, I will only use data from 2005 to 2008 due to the magnificent size of the datasets. There will be a sufficient amount of data to use in the project.
    </p>
<br> <br>
<p style="font-size:18px">In the follwing cell, I will download four dataset files into the notebook </p>

In [2]:
_2005_df = pd.read_csv('dataset/2005.csv')
_2006_df = pd.read_csv('dataset/2006.csv')
_2007_df = pd.read_csv('dataset/2007.csv')
_2008_df = pd.read_csv('dataset/2008.csv')

In [4]:
dfs = [_2005_df,_2006_df,_2007_df,_2008_df]
flight_df = pd.concat(dfs)

In [6]:
flight_df.shape

(28745461, 29)

In [8]:
flight_df.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  2005      1          28          5   1603.0        1605   1741.0   
1  2005      1          29          6   1559.0        1605   1736.0   
2  2005      1          30          7   1603.0        1610   1741.0   
3  2005      1          31          1   1556.0        1605   1726.0   
4  2005      1           2          7   1934.0        1900   2235.0   

   CRSArrTime UniqueCarrier  FlightNum  ... TaxiIn  TaxiOut  Cancelled  \
0        1759            UA        541  ...    4.0     23.0          0   
1        1759            UA        541  ...    6.0     15.0          0   
2        1805            UA        541  ...    9.0     18.0          0   
3        1759            UA        541  ...   11.0     10.0          0   
4        2232            UA        542  ...    5.0     10.0          0   

   CancellationCode  Diverted  CarrierDelay WeatherDelay NASDelay  \
0               NaN         0           0.0          0.0      0.0   
1               NaN         0           0.0          0.0      0.0   
2               NaN         0           0.0          0.0      0.0   
3               NaN         0           0.0          0.0      0.0   
4               NaN         0           0.0          0.0      0.0   

   SecurityDelay  LateAircraftDelay  
0            0.0                0.0  
1            0.0                0.0  
2            0.0                0.0  
3            0.0                0.0  
4            0.0                0.0  

[5 rows x 29 columns]

In [11]:
flight_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 28745461 entries, 0 to 7009727
Data columns (total 29 columns):
 #   Column             Dtype  
---  ------             -----  
 0   Year               int64  
 1   Month              int64  
 2   DayofMonth         int64  
 3   DayOfWeek          int64  
 4   DepTime            float64
 5   CRSDepTime         int64  
 6   ArrTime            float64
 7   CRSArrTime         int64  
 8   UniqueCarrier      object 
 9   FlightNum          int64  
 10  TailNum            object 
 11  ActualElapsedTime  float64
 12  CRSElapsedTime     float64
 13  AirTime            float64
 14  ArrDelay           float64
 15  DepDelay           float64
 16  Origin             object 
 17  Dest               object 
 18  Distance           int64  
 19  TaxiIn             float64
 20  TaxiOut            float64
 21  Cancelled          int64  
 22  CancellationCode   object 
 23  Diverted           int64  
 24  CarrierDelay       float64
 25  WeatherDelay     

### What is the structure of your dataset?

<p style="font-size:18px">The dataset `flight_df` contains 28745461 recoreds and 29 variables. Most of the variables are numaric, however, some of them their datatypes need to be converted to more aproprate datatypes.</p>

### What is/are the main feature(s) of interest in your dataset?
<p style="font-size:18px">My primary investigation and analysis will be focusing on figuring out the reasons behind the flight delays and cancelations in favor of measuring the punctuality of each flight and highlighting other flights' characteristics.</p>.

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

<p style="font-size:18px">II anticipate that variables associated with the delays in the departing and arriving time will play as good factors when associated with flight carriers and destinations to discover any hidden patterns. Not only this, but I think I can also use the cancellation code to find out the reasons behind the cancellations and link them to certain destinations and carriers.</p>

<a id='unidata'>   </a>
## Part II - Univariate Data Exploration

<a id='bivData'>   </a>
## Part III - Bivariate Data Exploration

<a id='multivData'>   </a>
## Part IV - Multivariate Data Exploration